### DISCUSSION TITLE
Custom software installation / ubuntu2004

### DISCUSSION THREAD

**Discussion by runnerImages** [actions/virtual-environments#8570] [Installed software]
I've added a custom software installation script that installs Terragrunt in the image build.
Script is also including a simple test (as presented below).
It occurred that terragrunt is not available in the scale set instances deployed from custom image.

It is very important for me to have it installed in the image (same as i.e. terraform and other tools).
I've seen some discussions about installing it in the pipeline where terragrunt is used, but unfortunately I can't use that solution due to organizational procedures.

Could you please let me know what other steps should I perform to add extra software into the image?

_agentrunners/images/linux/ubuntu2004.json_
```
"scripts": [
  "{{template_dir}}/scripts/installers/action-archive-cache.sh",
  "{{template_dir}}/scripts/installers/apt-common.sh",
  "{{template_dir}}/scripts/installers/customsoftwarelin.sh",
  "{{template_dir}}/scripts/installers/azcopy.sh",
``` 

_customsoftwarelin.sh_
```
#!/bin/bash -e
################################################################################
##  File:  customsoftware.sh
##  Desc:  Installs custom software
################################################################################

source $HELPER_SCRIPTS/install.sh

# Install Terragrunt
echo 'Terragrunt installation'
json=$(curl -s "https://api.github.com/repos/gruntwork-io/terragrunt/releases/latest")
downloadUrl=$(echo $json | jq -r ".assets | .[] | select(.name==\"terragrunt_linux_amd64\").browser_download_url")
curl $downloadUrl -4 -sL -o '/usr/local/bin/terragrunt'
chmod +x /usr/local/bin/terragrunt

# Test installation
echo 'Test Terragrunt installation'
terragrunt --version
```

**Comment by v-sergei-pyshnoi** [comment]
As I see, you already adds custom script to packer template. So for now you need to set building environment and start build process. Here is [step-by-step guide](https://github.com/actions/runner-images/blob/main/docs/create-image-and-azure-resources.md) with explaining how to do it.

**Reply by runnerImages** [reply]
Thank you, I'll check it! :)

**Comment by runnerImages** [comment]
Thank you @sergei-pyshnoi for help. I've found the problem which was not related to the software installation.
